src/ORM/Rule/ExistsIn.php

```diff
@@ -129,4 +146,28 @@ protected function _fieldsAreNull($entity, $source)
         }
         return $nulls === count($this->_fields);
     }
+
+    /**
+     * Checks whether or not the give entity fields are null and map to schema NULL
+     * or are not null and map to schema NOT NULL.
+     *
+     * @param \Cake\Datasource\EntityInterface $entity The entity to check.
+     * @param \Cake\ORM\Table $source The table to use schema from.
+     * @return bool
+     */
+    protected function _checkPartialSchemaNulls($entity, $source)
+    {
+        $schema = $source->schema();
+        foreach ($this->_fields as $field) {
+            $isNullable = $schema->isNullable($field);
+            $value = $entity->get($field);
+            if (!$isNullable && $value === null) {
+                return false;
+            }
+            if ($isNullable && $value !== null) {
+                return false;
```

Comment:
In that case `ExistsIn::_checkPartialSchemaNulls($entity, $source)` should return `false` and thus there should be no early return `true` in `ExistsIn::__invoke()` in line 114 but  `return $target->exists($conditions);` should still return true.

@chinpei215 Should I add a test case for the latter? Do you know of a fixture that fits, if I can't find one I'll need to create one.